freshrss: init at 1.18.1, tests and module

etu opened the pull request:
###### Motivation for this change
I've been using a module like this in my own configuration repo for quite some time, it worked fine through updates and things.

FreshRSS does some quite weird things in it's data folder. I'm certain that we don't want to even try to replicate that behavior.

So I'd rather just do the php setup and leave the installation to the user. There the user can select the database and such by itself and set it up as they wish as well.

This fixes #77246.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

etu committed:
freshrss: init at 1.18.1

etu committed:
nixos/freshrss: Init new module FreshRSS

etu committed:
nixos/tests: Init FreshRSS tests

SuperSandro2000 commented on pkgs/servers/freshrss/default.nix:
```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib, fetchFromGitHub, nixosTests }:
+
+let
+  pname = "FreshRSS";
+  version = "1.18.1";
+in
+stdenv.mkDerivation {
+  inherit pname version;
```
```suggestion
stdenv.mkDerivation rec {
  pname = "FreshRSS";
  version = "1.18.1";
```

etu replied:
No, I won't make a recursive attribute set.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133035` at 53186e90 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>freshrss</li>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/freshrss/default.nix:22:3:
```
   |
22 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

etu commented:
@ofborg build freshrss freshrss.tests

ajs124 commented:
See also #77246 and #77354

etu commented:
I'll resolve the release note conflicts whenever someone has approved the PR for merging, since it might take some time it's not worth to fix the same conflict over and over again.

aanderse commented on nixos/modules/services/web-apps/freshrss.nix:
```diff
@@ -0,0 +1,122 @@
+        StateDirectory = "freshrss";
+        WorkingDirectory = "/var/lib/freshrss";
+      };
+      script = ''
+        # Delete all but the "data" folder
+        ls | grep -v data | while read line; do rm -rf $line; done || true
+
+        # Copy all needed source files
+        cp -vr $(find ${pkgs.freshrss} -maxdepth 1 -mindepth 1 | grep -e '/cli' -e '/config' -e '/index' -e '/constants.php' -e '/app' -e '/extensions' -e '/p' -e '/lib') .
+
+        # Copy the user data template directory
+        cp -vr ${pkgs.freshrss}/data .
+
+        # Remove do-install.txt if we're already set up
+        if test -e data/config.php; then
+          rm data/do-install.txt
+        fi
+      '';
```
If I recall correctly it was possible to separate immutable code from variable state data. Setting a value for `DATA_PATH` in `constants.local.php` seemed to work for me, IIRC:

https://github.com/NixOS/nixpkgs/pull/77354/files#diff-3d624f87a5260006049b30c890415dc4b377695be2d403382b5809be573717daR14

ryantm replied:
This `Delete all but the "data" folder` thing seems scary. Perhaps you could use symlinks instead of copying.

etu committed:
freshrss: 1.18.1 -> 1.19.2

https://github.com/FreshRSS/FreshRSS/releases/tag/1.19.0
https://github.com/FreshRSS/FreshRSS/releases/tag/1.19.1
https://github.com/FreshRSS/FreshRSS/releases/tag/1.19.2

Stunkymonkey commented:
what is the current status here? I am interested in getting this merged.

etu commented:
I'm also interested in getting this merged, but due to life reasons I won't
have time to work on this in the foreseeable future.

If you're willing to pick up from where I've left it I'm fine with that. :)

On Sun, Jun 26, 2022, 19:39 Felix Bühler ***@***.***> wrote:

> what is the current status here? I am interested in getting this merged.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/133035#issuecomment-1166597743>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADQZEWMUC7NFKR6MZ7U2YTVRCIVRANCNFSM5BXTTRSA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
